substrate/frame/election-provider-multi-block/src/signed/mod.rs

```diff
@@ -970,4 +960,40 @@ impl<T: Config> Pallet<T> {
 		.defensive();
 		debug_assert_eq!(_res, Ok(to_slash));
 	}
+
+	/// Common logic for handling solution rejection - slash the submitter and try next solution
+	fn handle_solution_rejection(current_round: u32, reason: &'static str) {
+		if let Some((loser, metadata)) =
+			Submissions::<T>::take_leader_with_data(current_round).defensive()
+		{
+			// Slash the deposit
+			let slash = metadata.deposit;
+			let _res = T::Currency::burn_held(
+				&HoldReason::SignedSubmission.into(),
+				&loser,
+				slash,
+				Precision::BestEffort,
+				Fortitude::Force,
+			);
+			debug_assert_eq!(_res, Ok(slash));
+			Self::deposit_event(Event::<T>::Slashed(current_round, loser.clone(), slash));
+
+			// Try to start verification again if we still have submissions
+			if crate::Pallet::<T>::current_phase().is_signed_validation() &&
```

Comment:
idea: here we can actually double check that the inner value of `SignedValidation(x)` (`x`) is at least equal to `T::Pages`, or it is equal to zero.